lib/internal/repl/utils.js

```diff
@@ -87,7 +113,137 @@ function isRecoverableError(e, code) {
+      session.post('Runtime.evaluate', {
+        // Wrap the expression into brackets. Otherwise it might not be
+        // evaluated properly.
+        expression: `(${input})`,
+        throwOnSideEffect: true,
+        timeout: 333,
+        contextId: repl[contextSymbol],
+      }, (error, preview) => {
+        if (error) {
+          callback(error);
+        } else if (preview.result.value !== undefined) {
+          callback(null, inspect(preview.result.value, inspectOptions));
+        // Ignore EvalErrors, SyntaxErrors and ReferenceErrors. It is not clear
+        // where they came from and if they are recoverable or not. Other errors
+        // may be inspected.
+        } else if (preview.exceptionDetails &&
+                   (preview.result.className === 'EvalError' ||
+                     preview.result.className === 'SyntaxError' ||
+                     preview.result.className === 'ReferenceError')) {
+          callback(null, null);
+        } else if (preview.result.objectId) {
+          session.post('Runtime.callFunctionOn', {
+            functionDeclaration: `(v) => util.inspect(v, ${inspectedOptions})`,
+            objectId: preview.result.objectId,
+            arguments: [preview.result]
+          }, (error, preview) => {
+            if (error) {
+              callback(error);
+            } else {
+              callback(null, preview.result.value);
+            }
+          });
+        } else {
+          // No preview available / undefined is likely the outcome.
+          callback(null, 'undefined');
+        }
+      });
+    }, () => callback(new ERR_INSPECTOR_NOT_AVAILABLE()));
+  }
+
+  const showInputPreview = () => {
+    let line = repl.line.trim();
+
+    // Remove trailing semicolons. Otherwise it's not possible to inspect all
+    // input properly.
+    if (line.endsWith(';')) {
+      line = line.slice(0, line.length - 1);
+    }
+
+    // Do not preview if the command is buffered or if the line is empty.
+    if (repl[bufferSymbol] || line === '') {
+      return;
+    }
+
+    getPreviewInput(line, (error, inspected) => {
+      // Ignore the output if the value is identical to the current line and the
+      // former preview is not identical to this preview.
+      if ((line === inspected && lastPreview !== inspected) ||
+          inspected === null) {
+        return;
+      }
+      if (error) {
+        debug('Error while generating preview', error);
+        return;
+      }
+      // Do not preview `undefined` if colors are deactivated or explicitly
+      // requested.
+      if (inspected === 'undefined' &&
+          (!repl.useColors || repl.ignoreUndefined)) {
+        return;
+      }
+
+      preview = inspected;
+      lastPreview = preview;
+
+      // Limit the output to maximum 250 characters. Otherwise it becomes a)
+      // difficult to read and b) non terminal REPLs would visualize the whole
+      // output.
+      const maxColumns = MathMin(repl.columns, 250);
```

Comment:
It would be better, if we can take here the maximum length of the `tty` `column`? 🤔

Reply:
That is exactly what's done here. This is the inherited readline getter that accesses the streams columns in case a stream is set and returns Infinity otherwise. Printing more than 250 characters does not seem to be ideal in any case though. Previews should ideally be quickly accessible and there are cases where a stream might not have any columns. This addresses that problem.